### PR TITLE
71 save task to db

### PR DESCRIPTION
**Pull request by malsolec**
- Add Dagger 2 for dependency injection
- Add Task template Repository, with base methods create, get, delete
- Connect Task Create Activity with repository and save new task on click next button
- Add unit tests for task template repository
- Add activity tests for Task Create Actitivity
- Change dao for autoincrement, and change task duration time from string to int

### CODE REVIEW

**Comment by mikolevy** [comment]
I'm just about finishing my task with Codacy. Long story short:
- I marked all issues causing by potentially reasonable rules but connected with generated code or not relevant because of context as ignored
- I turned off some not reasonable rules
- I fixed some issues (import ordering, etc.)

In the result on my branch (which is copy of our master 3 days ago) there are only 3 issues left and after applying same approach to this PR there are only 5 new issues left (I think these ones are relevant but they can be fixed with my Codacy branch - because I've already did this :) )

File: Friendly-plans/app/src/main/java/dao/TaskTemplateRepository.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public long create(String name, int durationTime) {
+        TaskTemplate taskTemplate = new TaskTemplate();
+        taskTemplate.setName(name);
+        taskTemplate.setDurationTime(durationTime);
+
+        return daoSession.getTaskTemplateDao().insert(taskTemplate);
+    }
+
```
Comment (Artem3) [inline Friendly-plans/app/src/main/java/dao/TaskTemplateRepository.java]: How about another name: getId(Long id) ?

File: Friendly-plans/app/src/main/java/dao/TaskTemplateRepository.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public TaskTemplate get(Long id) {
+        return daoSession.getTaskTemplateDao().load(id);
+    }
+
```
Comment (Artem3) [inline Friendly-plans/app/src/main/java/dao/TaskTemplateRepository.java]: and getName(String taskName) ?

Comment (malsolec) [inline Friendly-plans/app/src/main/java/dao/TaskTemplateRepository.java]: Then name is ok in my opinion, we are not getting name but TaskTemplate.

File: Friendly-plans/app/src/main/java/dao/TaskTemplateRepository.java (hunk 1 above)
Comment (malsolec) [inline Friendly-plans/app/src/main/java/dao/TaskTemplateRepository.java]: Then name is ok in my opinion, we are not getting id but TaskTemplate.

**Comment by malsolec** [comment]
Hey, the test failed because of android orientation: horizontal or vertical. In vertical the next button is hide and it cannot be clicked. I will look how to fix it, sorry.

**Comment by malsolec** [comment]
Hey, I tried to fix it, it is not the screen orientation, I also add hiding a keyboard, and sleep, but it didn't help. No idea how to fix it for that moment. Any help?

**Comment by mikolevy** [comment]
Great :) I'll try look at this Travis problem during the weekend

**Pull request merged**